### PR TITLE
ember-cli: init at 5.3.0

### LINKED ISSUE #250506: Package request: ember-cli
**ember-cli nodePackage**

The Ember.js command line utility.

https://www.npmjs.com/package/ember-cli

If someone could give us some pointers we might actually have a crack at it ourselves ... @echus

**Metadata**

* homepage URL: https://cli.emberjs.com/release/
* source URL: https://github.com/ember-cli/ember-cli
* license: mit
* platforms: unix, linux, darwin

### PR DESCRIPTION
**Pull request by jfvillablanca**
## Description of changes
Resolves https://github.com/NixOS/nixpkgs/issues/250506

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jfvillablanca** [comment]
Result of `nixpkgs-review pr 260990` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ember-cli</li>
  </ul>
</details>

File: pkgs/by-name/em/ember-cli/package.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+let
+  pname = "ember-cli";
+  version = "5.3.0";
+in
+mkYarnPackage rec {
+  inherit pname version;
+
+  src = fetchFromGitHub {
+    owner = "ember-cli";
+    repo = pname;
```
Comment (eclairevoyant) [inline pkgs/by-name/em/ember-cli/package.nix:16]:
```suggestion
    repo = "ember-cli";
```
They are semantically different, this practice should be removed from nixpkgs.

File: pkgs/by-name/em/ember-cli/package.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+, mkYarnPackage
+, fetchFromGitHub
+, fetchYarnDeps
+}:
+
+let
+  pname = "ember-cli";
+  version = "5.3.0";
+in
+mkYarnPackage rec {
```
Comment (eclairevoyant) [inline pkgs/by-name/em/ember-cli/package.nix:11]: no point in having the `let` expr if we're going to have `rec` anyway, I'd remove the `let` tbh

File: pkgs/by-name/em/ember-cli/package.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    hash = "sha256-QgT2JFvMupJo+pJc13n2lmHMZkROJRJWoozCho3E6+c=";
+  };
+
+  packageJSON = ./package.json;
+
+  meta = with lib; {
+    homepage = "https://github.com/ember-cli/ember-cli";
+    description = "The Ember.js command line utility";
+    license = licenses.mit;
+    maintainers = with maintainers; [ jfvillablanca ];
```
Comment (eclairevoyant) [inline pkgs/by-name/em/ember-cli/package.nix:32]: please set `meta.mainProgram`

**Review by eclairevoyant — DISMISSED** [review]

File: pkgs/by-name/em/ember-cli/package.nix (hunk 2 above)
Comment (eclairevoyant) [reply pkgs/by-name/em/ember-cli/package.nix:11]: alternatively, `src` can be moved into the `let` block, then we can remove `rec`.

**Comment by SuperSandro2000** [comment]
@ofborg build ember-cli

**Comment by JamesWilmot** [comment]
Sweeeeet. Thanks for doing the hard yards on this!

I'll roll a flake to kick the tyres.

**Pull request merged**